CHANGELOG.md

```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.0.0] - 2023-06-05
+
+Support for both python 3.6 and 3.7 is removed.
```

Comment:
I believe there have been some more changes since the last release but if you don't feel like checking I can handle updating this when I make the release.

Reply:
Using commit [11cf9ef468a9f5d54d83275b9317492bf98c68ce](https://github.com/mivade/argparse_dataclass/commit/11cf9ef468a9f5d54d83275b9317492bf98c68ce) as the starting point the following changes were as follows:

+ Add`TypeError` for non-`Optional` `Union` typing with prompt to use custom `type`
+ Add support for `metavar` in metadata
+ `Literal` types as an alternative for fixed-choice fields
+ Prevent `Literal` and `choices` collision
+ Unit tests for:
  - union typing

I added the above points in `CHANGELOG.md`, feel free to add any that I may have missed!